amaranth_boards/tang_nano_9k.py

```diff
@@ -0,0 +1,135 @@
+import os
+import subprocess
+
+from amaranth.build import *
+from amaranth.vendor.gowin import *
+
+try:
+    from .resources import *
+except:
+    from resources import *
+
+
+__all__ = ["TangNano9kPlatform"]
+
+
+class TangNano9kPlatform(GowinPlatform):
+    part          = "GW1NR-LV9QN88PC6/I5"
+    family        = "GW1NR-9C"
+    default_clk   = "clk27" # or "OSC" to use on-chip oscillator
+    osc_frequency = 2500000 # Hz
+    board         = "tangnano9k"
+    resources     = [
+        Resource("clk27", 0, Pins("52", dir="i"),
+                 Clock(27e6), Attrs(IO_TYPE="LVCMOS33")),
+
+        *ButtonResources(pins="3 4",
+                         attrs=Attrs(IO_TYPE="LVCMOS33")),
+
+        # Pins 5 to 8 are JTAG (TMS, TCK, TDI, TDO) from BL702
+
+        # Pin 12 is VCC03_1V8
+        # LEDs are connected to 1V8, pins pull cathode to GND, when low
+        *LEDResources(pins="10 11 13 14 15 16", invert=True,
+                      attrs=Attrs(IO_TYPE="LVCMOS33")),
+
+        # Connects to BL702 UART1
+        UARTResource(0, rx="18", tx="17",
+            attrs=Attrs(PULL_MODE="UP", IO_TYPE="LVCMOS33")),
+
+        *SPIFlashResources(0,
```

Comment:
Is this also a PSRAM on the 9k? If so, it should be:
```suggestion
        *SPIFlashResources("psram", 0,
```